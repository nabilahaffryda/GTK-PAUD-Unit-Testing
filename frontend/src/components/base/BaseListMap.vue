<template>
  <v-container fluid class="px-4">
    <v-card flat>
      <v-container v-if="location && location.lat && location.lng">
        <v-toolbar elevation="0" dense>
          <v-list class="pb-0">
            <v-list-item class="pa-0">
              <v-list-item-content class="pa-0">
                <v-list-item-title class="title">
                  <h3>{{ title }}</h3>
                </v-list-item-title>
              </v-list-item-content>
            </v-list-item>
          </v-list>
        </v-toolbar>
        <div class="px-4 pb-2">
          <GmapMap
            ref="mapRef"
            :options="{
              zoomControl: true,
              mapTypeControl: false,
              scaleControl: false,
              streetViewControl: false,
              rotateControl: false,
              fullscreenControl: true,
              disableDefaultUi: false,
              gestureHandling: 'cooperative',
            }"
            :center="location"
            :zoom="zoom"
            :style="`width: ${width}; height: ${height}`"
          >
            <gmap-marker
              v-for="(m, index) in markers"
              :icon="$imgUrl(icon)"
              :position="m"
              :key="index"
              :clickable="true"
              @click="onInfoWindow(m)"
            />
            <GmapInfoWindow
              :options="{ maxWidth: 300 }"
              :position="iWindow.position"
              :opened="iWindow.open"
              @closeclick="iWindow.open = false"
            >
              <div v-html="iWindow.template"></div>
            </GmapInfoWindow>
          </GmapMap>
        </div>
      </v-container>
    </v-card>
  </v-container>
</template>

<script>
export default {
  name: 'ListMap',
  props: {
    title: {
      type: String,
      default: '',
    },
    location: {
      type: Object,
      default: () => {},
    },
    zoom: {
      type: Number,
      default: 13,
    },
    width: {
      type: String,
      default: '100%',
    },
    height: {
      type: String,
      default: '300px',
    },
    markers: {
      type: Array,
      default: () => [],
    },
    icon: {
      type: String,
      default: 'icn-map-lokasi.png',
    },
  },
  data() {
    return {
      iWindow: { open: false },
    };
  },
  methods: {
    onInfoWindow() {},
  },
};
</script>

<style scoped>
.link {
  white-space: normal;
  cursor: pointer;
}
</style>
