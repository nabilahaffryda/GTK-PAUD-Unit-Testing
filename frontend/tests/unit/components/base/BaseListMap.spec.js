import { mount, createLocalVue } from "@vue/test-utils";
import BaseListMap from "@/components/base/BaseListMap.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';
import VueRouter from 'vue-router';

Vue.use(Vuetify)
const localVue = createLocalVue();
localVue.use(VueRouter)
const router = new VueRouter();

describe('BaseListMap.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    function wrapperFactory({ } = {}) {
        return mount(BaseListMap, {
            localVue,
            vuetify,
            router,
        });
    }
    test('call onInfoWindow method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.onInfoWindow = jest.fn();
        wrapper.vm.onInfoWindow();
        expect(wrapper.vm.onInfoWindow.mock.calls.length).toBe(1);
    })
})