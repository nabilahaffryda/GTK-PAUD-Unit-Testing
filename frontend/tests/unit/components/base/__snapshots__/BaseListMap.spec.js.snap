// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseListMap.vue render list map 1`] = `
<div class="container px-4 container--fluid">
  <div class="v-card v-card--flat v-sheet theme--light"></div>
</div>
`;
